ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-AllergyIntolerance.xsl

```diff
@@ -0,0 +1,253 @@
+    xmlns="http://hl7.org/fhir"
+    xmlns:util="urn:hl7:utilities" 
+    xmlns:f="http://hl7.org/fhir" 
+    xmlns:xd="http://www.oxygenxml.com/ns/doc/xsl"
+    xmlns:nf="http://www.nictiz.nl/functions" 
+    xmlns:uuid="http://www.uuid.org"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform" 
+    version="2.0">
+    
+    <xsl:output method="xml" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+    
+    <xd:doc scope="stylesheet">
+        <xd:desc>Converts ADA [...] to FHIR [...] conforming to profile [...]</xd:desc>
+    </xd:doc>
+    
+    <xd:doc>
+        <xd:desc>Create a nl-core-AllergyIntolerance instance as a AllergyIntolerance FHIR instance from ADA allergie_intolerantie.</xd:desc>
+        <xd:param name="in">ADA element as input. Defaults to self.</xd:param>
+        <xd:param name="subject">The AllergyIntolerance.patient as ADA element or reference.</xd:param>
+    </xd:doc>
+    <xsl:template name="nl-core-AllergyIntolerance" mode="nl-core-AllergyIntolerance" match="allergie_intolerantie" as="element(f:AllergyIntolerance)">
+        <xsl:param name="in" as="element()?" select="."/>
+        <xsl:param name="subject" select="patient/*" as="element()?"/>
+        
+        <xsl:for-each select="$in">
+            <AllergyIntolerance>
+                <xsl:call-template name="insertLogicalId"/>
+                
+                <meta>
+                    <profile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-AllergyIntolerance"/>
+                </meta>
+         
+                <xsl:for-each select="allergie_status">
+                    <xsl:choose>
+                        <xsl:when test="(./@code = 'active') 
+                            or (./@code = 'obsolete') 
+                            or (./@code = 'completed')">
+                            <clinicalStatus>
+                                <xsl:call-template name="code-to-CodeableConcept">
+                                    <xsl:with-param name="in" select="."/>
+                                    <xsl:with-param name="codeMap" as="element()*">
+                                        <map inCode="active" inCodeSystem="2.16.840.1.113883.5.14" code="active" codeSystem="http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical" displayName="Active"/>
+                                        <map inCode="obsolete" inCodeSystem="2.16.840.1.113883.5.14" code="inactive" codeSystem="http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical" displayName="Inactive"/>
+                                        <map inCode="completed" inCodeSystem="2.16.840.1.113883.5.14" code="resolved" codeSystem="http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical" displayName="Resolved"/>
+                                    </xsl:with-param>
+                                </xsl:call-template>
+                                <xsl:call-template name="code-to-CodeableConcept">
```

Comment:
The profile indeed does not require the zib terminology to be included. It is however not forbidden. Perhaps I thought that it would be good to also include the zib terminology code. On the other hand, more bytes without a real function. 
I will remove it.

Reply:
Well ... I don't care about the bytes and normally I would agree that extra terminology is a good thing. But the example should demonstrate how the profile is to be used, so adding the zib terminology here is confusing (n = 1).